tests/python/gaia-ui-tests/gaiatest/apps/system/regions/time_picker.py

```diff
@@ -23,15 +25,18 @@ class TimePicker(Base):
     def __init__(self, marionette):
         Base.__init__(self, marionette)
         self.marionette.switch_to_frame()
-        self.wait_for_element_displayed(*self._current_element(*self._hour_picker_locator))
+        element = Wait(self.marionette).until(
+            expected.element_present(*self._current_element(*self._hour_picker_locator)))
+        Wait(self.marionette).until(expected.element_displayed(element))
 
-        # TODO: wait for the time piker to fade in Bug 1038186
+        # TODO: wait for the time picker to fade in Bug 1038186
         time.sleep(2)
 
     def tap_done(self):
         self.marionette.find_element(*self._done_button_locator).tap()
-        self.wait_for_element_not_displayed(*self._time_picker_locator)
-        # TODO: wait for the time piker to fade out Bug 1038186
+        Wait(self.marionette).until(
+            expected.element_not_displayed(*self._time_picker_locator))
```

Comment:
If this is present/visible before the tap then find it before and wait for it to be not displayed after.